Python_Scripts/script_regroup_models.py

```diff
@@ -0,0 +1,337 @@
+    final_folder = os.path.join(DIR_MODELS_REGROUPED, model_details['short_id'])
+    final_file_name = os.path.join(final_folder, sbml_file_name)
+
+    # ifwe have no SED-ML file, the model is a biomodels model consisting of
+    # only one SBML sheet. Movi this file to the new location
+    if model_details['sedml_path'] == '':
+        shutil.copy(model_details['sbml_path'], final_file_name)
+        return final_file_name
+
+    # if the model is from the JWS model collection, we must applay the changes
+    # from the SED-ML file
+    sedml_file = libsedml.readSedML(model_details['sedml_path'])
+    sedml_task = sedml_file.getTask(model_details['sedml_task'])
+    sbml_file = libsbml.readSBML(model_details['sbml_path'])
+    sbml_model = sbml_file.getModel()
+
+    # short consistency check that SBML model and SED-ML task fit together
+    assert sbml_file_name.split('.')[0] == sedml_task.getModelReference()
+    sedml_submodel = sedml_file.getModel(sedml_task.getModelReference())
+
+    logfile = open('sedml_change.log', 'a')
+    for change in sedml_submodel.getListOfChanges():
+        # We iterate over all changes for this model in the SED-ML
+        target = change.getTarget()
+        value = change.getNewValue()
+
+        # decide for species or parameter
+        div = (target.split('[')[0]).split(':')[4]
+        target_id = target.split('\'', )[1]
+
+        if div == 'species':
+            try:
+                sbml_model.getSpecies(target_id).initial_concentration = \
+                    float(value)
+            except AttributeError:
+                if sbml_model.getSpecies(target_id) is None:
+                    logfile.write(f'Trying to change initial concentration of species '
+                          f'{target_id} via SED-ML file, but this species does '
+                          f'not exist in the current SBML model '
+                          f'({sbml_file_name}). Omitting!\n')
+        elif div == 'parameter':
+            try:
+                sbml_model.getParameter(target_id).value = float(value)
+            except AttributeError:
+                if sbml_model.getParameter(target_id) is None:
+                    logfile.write(f'Trying to change sbml_model parameter {target_id}, '
+                          f'via SED-ML file, but this parameter does not exist '
+                          f'in the current SBML model ({sbml_file_name}).'
+                          f'Omitting!\n')
+        elif div == 'compartment':
+            sbml_model.getCompartment(target_id).size = float(value)
+        else:
+            logfile.write(f'Trying to change a {div} in sbml_model '
+                  f'{sbml_file_name}, which is currently not supported.'
+                  f'Omitting!\n')
+
+    # save changed sbml files and the accompaniying sedml file
+    libsbml.writeSBMLToFile(sbml_model.getSBMLDocument(), final_file_name)
+    logfile.close()
+    return final_file_name
```

Comment:
looks reasonable. I would say shortly check whether the number of models accepted this way changed compared to the previous implementation. If less, sth may be wrong here.

Reply:
More sbml files accepted know, although wrapping works for less models. This is somewhat fishy.
In total less benchmark models, due to another way of grouping.